client/signup/steps/domains/index.jsx

```diff
@@ -392,9 +392,17 @@ export class RenderDomainsStep extends Component {
 		}
 	};
 
-	submitWithDomain = ( { googleAppsCartItem, shouldHideFreePlan = false, signupDomainOrigin } ) => {
+	submitWithDomain = ( {
+		googleAppsCartItem,
+		shouldHideFreePlan = false,
+		signupDomainOrigin,
+		suggestion,
+	} ) => {
 		const { step } = this.props;
-		const { suggestion } = step;
+
+		if ( step.suggestion ) {
```

Comment:
@alshakero can I safely do this? For the existing flows, `step.suggestion` should work but it somehow isnt working for our flow so I had to pass it from line 271

Reply:
> `step.suggestion` should work but it somehow isnt working for our flow

@alshakero may know better, but it looks like we should investigate (and fix) the expected behavior

Reply:
Frankly I can't tell you if this is safe, what do you mean by not working, its not being submitted?

Reply:
Without this change it is `null`  and then it is null in providedDependencies too, which makes the domain not appear in checkout later.

Reply:
I tried and honestly I can't reproduce. Anyways, don't be blocked by this but please give it another shot.

Reply:
Now that this PR is merged, can we open a follow-up to clean this up?

Reply:
@ciampo I am not sure what do you mean by cleanup? I suspect this is the issue of the domain step not properly accounting for a situation when a domain step is launched AFTER website being created and the domain being added to the wrong cart (anonymous vs site specific). I honestly do not know how to properly fix it and account for other quirks of stepper.

Reply:
I'll try to debug this a bit more to maybe find more elegant solution